helpers/auth/authentication.py

```diff
@@ -133,21 +133,16 @@ def wrapper(*args, **kwargs):
                         user = User.query.filter_by(email=email).first()
                     except Exception:
                         raise GraphQLError("The database cannot be reached")
-                    user_values = response['values']
-                    if len(user_values):
-                        for value in user_values:
-                            if user.email == value["email"]:
-                                if value['location'] and not user.location:
-                                    user_location = value['location'][
-                                        'name'
-                                    ] or "Nairobi"
-                                    user.location = user_location
-                                    user.save()
-                                    check_and_add_location(user_location)
-                    else:  # pragma: no cover
-                        if not user.location:
-                            user.location = "Nairobi"
-                            user.save()
+                    for value in response['values']:
+                        if user.email == value["email"]:
+                            if value['location'] and not user.location:
+                                user_location = value['location']['name']
+                                user.location = user_location
+                                user.save()
+                                check_and_add_location(user_location)
+                    if not user.location:  # pragma: no cover
+                        user.location = "Nairobi"
+                        user.save()
```

Comment:
@rajeman we would still want to `check_and_add_location` here incase the `Nairobi` location does not exist.

Reply:
@joshuaocero This has been taken care of.